### PR TITLE
Set up Associations

### PR DESCRIPTION
**Pull request by Zeph180**
- [ ] - [ ]  Create a class Classroom
- [ ]  Create the has-many/belongs-to relationship between Classroom and Student
- [ ]  Create the has-many side (a classroom has many students).
- [ ]  Create the belongs-to side (a student belongs to a classroom).
- [ ] Create a class Book
- [ ] Setters and getters for instance variables (remember about attr_accessor).
- [ ] Create a class Rental
- [ ] Setter and getter for @date
- [ ] Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
- [ ] Create the has-many side of Book and Rental (a book has many rentals).
- [ ] Create the belongs-to side of Rental and Book (a rental belongs to a book).
- [ ] Create the has-many side of Person and Rental (a person has many rentals).
- [ ] Create the belongs-to side of Rental and Person (a rental belongs to a person).
- [ ] Modify the constructor of Rental so Book and Person are set in

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Rental
+  attr_accessor :date
```
Comment (VICTOR-CHUKWU) [inline rental.rb]: - [ ] kindly make book and person accessible too according to project requirement

File: person.rb [hunk 2]
```diff
@@ -29,6 +30,14 @@ def correct_name
   def of_age?
     @age >= 18
   end
+
+  def add_rental(rental)
+    @rentals << rental
```
Comment (VICTOR-CHUKWU) [inline person.rb]: - [ ] here you have to create a new instance of Rentals, remember in your rental you are already pushing to person.rental, so your code should look like this `Rental.new(date, book, self)`

File: person.rb [hunk 3]
```diff
@@ -14,6 +14,7 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
```
Comment (VICTOR-CHUKWU) [inline person.rb:17]: - [ ] kindly make rentals accessible :+1:

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
```
Comment (VICTOR-CHUKWU) [inline book.rb]: - [x] you should create a new instance of rentals, remember you already pushed to rental.book in your rental class your code should look like this `Rental.new(date, self, person)`

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @Zeph180  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Olamarx — APPROVED** [review]
Hi @Zeph180, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Linnters passed ✔️
- Readme file professional ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**